packages/io/stl-serializer/CSGToStla.js

```diff
@@ -1,50 +1,46 @@
-const { ensureManifoldness } = require('@jscad/io-utils')
-const {isCSG} = require('@jscad/csg')
+const { geometry } = require('@jscad/modeling')
 
-// objects must be an array of CSG objects (with polygons)
+// objects must be an array of 3D geomertries (with polygons)
 const serializeText = (objects, options) => {
-  options.statusCallback && options.statusCallback({progress: 0})
+  options.statusCallback && options.statusCallback({ progress: 0 })
 
   let result = `solid JSCAD
 ${convertToStl(objects, options)}
 endsolid JSCAD
 `
-  options.statusCallback && options.statusCallback({progress: 100})
+  options.statusCallback && options.statusCallback({ progress: 100 })
   return [result]
 }
 
 const convertToStl = (objects, options) => {
   let result = []
-  objects.forEach(function (object, i) {
+  objects.forEach((object, i) => {
     result.push(convertToFacets(object, options))
-    options.statusCallback && options.statusCallback({progress: 100 * i / objects.length})
+    options.statusCallback && options.statusCallback({ progress: 100 * i / objects.length })
   })
   return result.join('\n')
 }
 
 const convertToFacets = (object, options) => {
   let result = []
-  object.polygons.forEach(function (p, i) {
-    result.push(convertToFacet(p))
+  let polygons = geometry.geom3.toPolygons(object)
+  polygons.forEach((polygon, i) => {
+    result.push(convertToFacet(polygon))
   })
   return result.join('\n')
 }
 
-const vector3DtoStlString = (v) => {
-  return `${v._x} ${v._y} ${v._z}`
-}
+const vector3DtoStlString = (v) => `${v[0]} ${v[1]} ${v[2]}`
```

Comment:
perhaps since now we use array-like data, it would make more sense for a string representation to also match that ?

Reply:
I'll leave this for now. There's still room for improvements in the code.